expr/match.go

```diff
@@ -0,0 +1,62 @@
+package expr
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/google/nftables/binaryutil"
+	"github.com/mdlayher/netlink"
+	"golang.org/x/sys/unix"
+)
+
+// See https://git.netfilter.org/libnftnl/tree/src/expr/match.c?id=09456c720e9c00eecc08e41ac6b7c291b3821ee5#n30
+type Match struct {
+	Name string
+	Rev  uint32
+	Info []byte
+}
+
+func (e *Match) marshal() ([]byte, error) {
+	// Per https://git.netfilter.org/libnftnl/tree/src/expr/match.c?id=09456c720e9c00eecc08e41ac6b7c291b3821ee5#n38
+	name := e.Name
+	// leave room for trailing \x00; this looks like a weak spot in libnftnl
```

Comment:
Can you elaborate on what weak spot you see?

AFAICT, `libnftnl` uses:
```c
		snprintf(mt->name, sizeof(mt->name), "%.*s", data_len,
			 (const char *)data);
```

And `snprintf` is documented to write at most `sizeof(mt->name)` bytes, including the terminating null byte.

So the size limit here is 29 bytes, including the terminating null byte.

But the Go code you have permits 29 bytes *plus* an extra null byte.

Reply:
Ah, I missed the part that `snprintf` ensures to correctly terminate with `\x00`. So my comment is incorrect and I'll remove it; instead, I will "trim" this to 29 octets 😁 plus terminating zero octet.